go/adbc/driver/snowflake/connection.go

```diff
@@ -18,12 +18,14 @@
 package snowflake
 
 import (
+	"cmp"
 	"context"
 	"database/sql"
 	"database/sql/driver"
 	"fmt"
 	"io"
 	"regexp"
+	"slices"
```

Comment:
the packages `cmp` and `slices` were only added in `go1.21` so either we need to bump our minimum version or we need to replace the usages of these.

@lidavidm what do you think?

Reply:
I'd be fine with bumping to 1.21 as our minimum version

Reply:
Looks like 1.20 is EOL. Could someone create a separate PR to bump?

Reply:
I'll take care of that.

@ryan-syed after we merge the bump, I'll comment here so you can rebase.

Reply:
@zeroshade gentle reminder to check if the minimum version has been bumped up as this PR has been open for a while now.

Reply:
We ran into some issues with the packaging scripts when trying to bump the version. I'm looking into it with help from @kou

Reply:
I pushed a merge commit here.

Reply:
you beat me to it @lidavidm! lol